components/settings/actions/ImportExportHLS.js

```diff
@@ -0,0 +1,187 @@
+            simConfigPath: "",
+            simConfigModuleName: "",
+            simConfigVariable: "simConfig",
+            modFolder: "",
+            loadMod: '',
+            compileMod: false,
+            explorerDialogOpen: false,
+            explorerParameter: "",
+            exploreOnlyDirs: false,
+            filterFiles: false,
+            netParamsHovered: 'hidden'
+        }
+    }
+
+    componentDidUpdate() {
+        if (this.props.actionRequired && !this.state.actionExecuted) {
+            this.performAction()
+        }
+        else if (!this.props.actionRequired && this.state.actionExecuted) {
+            this.setState({actionExecuted: false})
+        }
+    }
+
+    performAction() {
+        if (this.props.requestID==5) {
+            var action = 'netpyne_geppetto.exportHLS';
+            var message = 'EXPORTING MODEL';
+            this.props.performAction(action, message, this.state)
+        }
+        else if (this.state.loadMod===undefined) { //this is cause of the warning (if mod in SelectField is not selected)
+        }
+        else if (this.state.loadMod==='') {
+            this.setState({loadMod: undefined, actionExecuted: true})
+            this.props.performAction('abort')
+        }
+        else {
+            var action = 'netpyne_geppetto.importModel';
+            var message = 'IMPORTING MODEL';
+            this.props.performAction(action, message, this.state)
+            this.setState({actionExecuted: true})
+        }
+    }
+
+    showExplorerDialog(explorerParameter, exploreOnlyDirs, filterFiles) {
+        this.setState({ explorerDialogOpen: true, explorerParameter: explorerParameter, exploreOnlyDirs: exploreOnlyDirs, filterFiles: filterFiles })
+    }
+
+    closeExplorerDialog(fieldValue) {
+        var newState = { explorerDialogOpen: false };
+        if (fieldValue) {
+            var fileName = fieldValue.path.replace(/^.*[\\\/]/, '');
+            var fileNameNoExtension = fileName.replace(/\.[^/.]+$/, "");
+            var path = fieldValue.path.split(fileName).slice(0, -1).join('');
+            switch (this.state.explorerParameter) {
+                case "netParamsPath":
+                    newState["netParamsPath"] = path;
+                    newState["simConfigPath"] = path;
+                    newState["netParamsModuleName"] = fileNameNoExtension;
+                    newState["simConfigModuleName"] = fileNameNoExtension;
+                    break;
+                case "simConfigPath":
+                    newState["simConfigPath"] = path;
+                    newState["simConfigModuleName"] = fileNameNoExtension;
+                    break;
+                case "modFolder":
+                    newState["modFolder"] = fieldValue.path;
+                    break;
+                default:
+                    throw ("Not a valid parameter!");
+            }
+        }
+        this.setState(newState);
+    }
+
+    render() {
+        var importModFiles = (
```

Comment:
sure, but we also have to keep in mind code readability. In the future we will put on top of this code, options for .mat and .pkl, and probably a switch, so maybe it would be useful to have the importModFile hanging on top and place it in the right place for each option.

Reply:
I agree but I wouldn't keep this change thinking that it may be needed in the future. There are so many things that can change in the future :)